modules/issue_tracker/php/edit.class.inc

```diff
@@ -0,0 +1,859 @@
+            [
+                'issueID'  => $issueID,
+                'uid'      => $user->getUsername(),
+                'assignee' => $changed_assignee,
+            ]
+        );
+
+        $msg_data['url']         = $baseurl .
+            "/issue_tracker/issue/" . $issueID;
+        $msg_data['issueID']     = $issueID;
+        $msg_data['currentUser'] = $user->getUsername();
+
+        foreach ($issue_change_emails as $email) {
+            $msg_data['firstname'] = $email['firstname'];
+            Email::send($email['Email'], 'issue_change.tpl', $msg_data);
+        }
+    }
+
+    /**
+     * Validate PSCID and Visit Label
+     *
+     * @param array $values values to be validated
+     * @param \User $user   the user requesting the change
+     *
+     * @return array|\LORIS\Http\Response\JSON\InternalServerError
+     */
+    function validateInput(array $values, \User $user)
+    {
+        $db         = $this->loris->getDatabaseConnection();
+        $pscid      = ($values['PSCID'] ?? null);
+        $visitLabel = ($values['visitLabel'] ?? null);
+        $centerID   = ($values['centerID'] ?? null);
+        $result     = [
+            'PSCID'             => $pscid,
+            'visit'             => $visitLabel,
+            'centerID'          => $centerID,
+            'candID'            => null,
+            'sessionID'         => null,
+            'isValidSubmission' => true,
+            'invalidMessage'    => null,
+        ];
+
+        if (isset($result['PSCID'], $result['centerID'])) {
+            $validCenter = $db->pselectOne(
+                "
+                SELECT
+                    RegistrationCenterID = :center_id as CenterID
+                FROM
+                    candidate
+                WHERE
+                    PSCID = :psc_id
+            ",
+                [
+                    "center_id" => $result['centerID'],
+                    "psc_id"    => $result['PSCID'],
+                ]
+            );
+            if (!$validCenter) {
+                $validCenter = $db->pselectOne(
+                    "
+                    SELECT
+                        EXISTS (
+                            SELECT
+                                *
+                            FROM
+                                session s
+                            JOIN
+                                candidate c
+                            ON
+                                c.CandID = s.CandID
+                            WHERE
+                                s.CenterID = :center_id AND
+                                c.PSCID = :psc_id
+                        )
+                ",
+                    [
+                        "center_id" => $result['centerID'],
+                        "psc_id"    => $result['PSCID'],
+                    ]
+                );
+            }
+            if (!$validCenter) {
+                error_log('PSCID and Center ID do not match a valid session!');
```

Comment:
I don't see why this is going to the error log, if it needs to it should be using `$this->logger`